### PR TITLE
Adds round end to status tab

### PR DESCRIPTION
**Pull request by BeagleGaming1**
# About the pull request

Lets ghosts and new players (lobby) see time to server restart

# Explain why it's good for the game

Lets you see if the automatic round-end timer ran out, and if the server is delayed
# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog
:cl:
add: Ghosts and players in lobby can see the time to the server restarting in the status tab
/:cl:

### CODE REVIEW

File: code/modules/mob/new_player/new_player.dm [hunk 1]
```diff
@@ -490,6 +490,15 @@
 	. += ""
 	. += "Game Mode: [GLOB.master_mode]"
 
+	if(SSticker.current_state == GAME_STATE_FINISHED)
+		if(SSticker.mode.round_end_time == 0)
+			. += "Time To Restart: Ongoing Vote [SSticker.delay_end ? "(DELAYED)" : ""]"
+			return
+		if(SSticker.mode.round_end_time < 0)
+			. += "Time To Restart: SOON [SSticker.delay_end ? "(DELAYED)" : "(RESTART NEEDED)"]"
```
Comment (realforest2001) [inline code/modules/mob/new_player/new_player.dm:498]:
```suggestion
			. += "Time To Restart: SOON [SSticker.delay_end ? "(DELAYED)" : "(MANUAL RESTART NEEDED)"]"
```
Assuming I interpreted this correctly, worth clarifying it needs to be done manually.

File: code/modules/mob/dead/observer/observer.dm [hunk 2]
```diff
@@ -1090,6 +1090,14 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
...
 			. += "Evacuation: [eta_status]"
 
+	if(SSticker.current_state == GAME_STATE_FINISHED)
+		if(SSticker.mode.round_end_time == 0)
+			. += "Time To Restart: Ongoing Vote [SSticker.delay_end ? "(DELAYED)" : ""]"
+			return
+		if(SSticker.mode.round_end_time < 0)
+			. += "Time To Restart: SOON [SSticker.delay_end ? "(DELAYED)" : "(RESTART NEEDED)"]"
+			return
+		. += "Time To Restart: [(SSticker.roundend_restart_delay + (SSticker.mode.round_end_time - world.time)) / 10 ]s [SSticker.delay_end ? "(DELAYED)" : ""]"
```
Comment (realforest2001) [inline code/modules/mob/dead/observer/observer.dm:1100]:
```suggestion
	if(SSticker.current_state == GAME_STATE_FINISHED)
		var/round_end_timer = "Time To Restart: [(SSticker.roundend_restart_delay + (SSticker.mode.round_end_time - world.time)) / 10 ]s [SSticker.delay_end ? "(DELAYED)" : ""]"
		if(SSticker.mode.round_end_time == 0)
			round_end_timer = "Time To Restart: Ongoing Vote [SSticker.delay_end ? "(DELAYED)" : ""]"
		if(SSticker.mode.round_end_time < 0)
			round_end_timer = "Time To Restart: SOON [SSticker.delay_end ? "(DELAYED)" : "(RESTART NEEDED)"]"
		. += round_end_timer
```
Optional suggestion.

**Review by realforest2001 — CHANGES_REQUESTED** [review]

**Review by realforest2001 — APPROVED** [review]

**Comment by BeagleGaming1** [comment]
Bruh its like 10s off on live

**Review by morrowwolf — CHANGES_REQUESTED** [review]
Allegedly weirdness with the timer during TM

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Pull request closed without merging**